uiowa/uiowa
# Education Migration

Resolves #3899 

- [x] Migrate all existing articles. (total count: 609)
- [x] Maintain published/unpublished status
- [x] Tags, Affiliations are migrated to Tags. (Might have a list of consolidation for us to map)
- [x] Source Title -> Organization
- [x] Source Link -> Source Link
- [x] Image -> Featured Image (site-wide default)
- [x] Caption -> stripped and truncated to caption
- [x] Body summary -> field_teaser
- [x] Byline -> Prepend to body
- [x] Short Description -> Prepend to body (uids-component--light-intro)
- [x] body -> body (convert btn-primary to bttn bttn--caps bttn--primary, 
- [x] body -> body pull-quote to blockquote)
- [x] Create redirects from old articles.
- [x] Provide initial list of potential broken links.





# How to test



Sync the d7 instance locally, and start up the stack
`blt ds --site=education.uiowa.edu`
`drush @education.local en education_migrate`
`drush @education.local uli /admin/config/system/sitenow-migrate`
Complete the config form using "https://education.uiowa.edu" as the base URL
Run the migrations. -- Will likely need to limit the articles migration using something like `drush @education.local mim education_articles --limit=200`, otherwise you may run into a memory issue. (Although! I think all the errors have been worked out, where even if you error out you can re-run it without issues.)
At the end of an import, a lengthy terminal log will contain messaging regarding all articles with suspected broken links (ie relative `node/#` links which are no longer accurate). This report should cover all migrated pages--if multiple imports were run due to the memory issues, only the final report is needed.
Check things out and see that everything looks awesome.
Check that article summaries do not contain any `&#13;` characters (carriage returns that were not properly removed or converted).
Check that article redirects from the previous D7 alias work as expected.

Note for the pull quotes: The following nodes on the existing site contain pull quotes. The first two each contain an instance of a `div` pull quote which consist of an image and caption. These have been converted to images with captions in the migration, and not pull quotes. The rest are instances of `p` pull quotes, which have been converted directly to `blockquote` in the migration.
| entity_id |
|------------|
2041	
2536	
6521	
8586	
8636	
11651	


Some nids from the D7 site which contained a byline field value:

| entity_id	| field_article_byline_value	|
|--------------|----------------------------|
961	| By Andrea Zeek	
2041 | 	By Mikaela Parrick	
2146 | 	By Andy Goodell	
2241 | 	By Andrea Zeek	
2266 | 	By Mikaela Parrick	
2306 | 	By Sean Thompson	
2536 | 	By Andy Goodell	
2806 | 	By Andy Goodell	
2831 | 	By Andy Goodell	
3076 | 	By Andy Goodell	

Some nids which used the short description field:

| entity_id | 	field_header_short_description_value	
|----------|------------------------------------------------------------|
9776 | 	For UI REACH graduate Andrew Sukup, the sky is the...	
9786 | 	UI alumnus Jake Chung plants seeds of hope in comm...	
9851 | 	Iowa City New Horizons band members share love of ...	
9896 | 	Rusty Barceló serves as mentor for generations of ...	
10191 | 	Collaboration, creativity lead to new exhibit at T...	
11191 | 	1.2 million Iowans live in rural communities.	
11201 | 	Helping all students experience safe and supportiv...	
12296 | 	The UI trains educators and administrators who are...	
12656 | 	 This conceptual rendering from CMBA Architects sho...	


And nids from D7 where buttons were present:

| entity_id	 |
|-----------|
679	
3491	
15271	
15276	
15411
